go-controller/pkg/ovn/controller/admin_network_policy/metrics.go

```diff
@@ -0,0 +1,130 @@
+package adminnetworkpolicy
+
+import (
+	"fmt"
+
+	libovsdbutil "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/util"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	anpapi "sigs.k8s.io/network-policy-api/apis/v1alpha1"
+)
+
+/*var metricANPRuleCount = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	Namespace: metrics.MetricOvnkubeNamespace,
+	Subsystem: metrics.MetricOvnkubeSubsystemController,
+	Name:      "admin_network_policy_rules", // doing a sum across all ANPs will give the absolute count in the cluster
+	Help:      "The total number of rules in a given admin network policy in the cluster"},
+	[]string{
+		"direction", // direction is either "ingress" or "egress"; so cardinality is max 2 for this label
+		"action",    // action is either "Pass" or "Allow" or "Deny"; so cardinality is max 3 for this label
+	},
+)
+
+var metricBANPRuleCount = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	Namespace: metrics.MetricOvnkubeNamespace,
+	Subsystem: metrics.MetricOvnkubeSubsystemController,
+	Name:      "baseline_admin_network_policy_rules",
+	Help:      "The total number of rules in a given baseline admin network policy in the cluster"},
+	[]string{
+		"direction", // direction is either "ingress" or "egress"; so cardinality is max 2 for this label
+		"action",    // action is either "Allow" or "Deny"; so cardinality is max 2 for this label
+	},
+)*/
+
+// Descriptors used by the ANPControllerCollector below.
+var (
+	anpRuleCountDesc = prometheus.NewDesc(
+		"admin_network_policy_rules",
+		"The total number of rules across all admin network policies in the cluster",
+		[]string{"direction", "action"}, nil,
+	)
+)
+
+// ANPControllerCollector implements the Collector interface.
+type ANPControllerCollector struct {
+	ANPController *Controller
+}
+
+func (c *Controller) initMetricsCollector() {
+	reg := prometheus.NewPedanticRegistry()
```

Comment:
You don't need a new registry if all metrics are registered with the global registry.